### PR TITLE
Add the ability to specify pex_root on command line

### PR DESCRIPTION
**Pull request by digwanderlust**
* add pex_root option
* fix some places where ~/.pex was hard coded

### CODE REVIEW

File: pex/pex_info.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class PexInfo(object):
   requirements: list                # list of requirements for this environment
 
   # Environment options
-  pex_root: ~/.pex                   # root of all pex-related files
+  pex_root: eg: ~/.pex               # root of all pex-related files
```
Comment (kwlzn) [inline pex/pex_info.py]: shouldn't this actually include the type information per the format of the other lines, i.e. 'string'? maybe something like:

```pex_root: string, default: ~/.pex                              # root of all pex-related files```

File: pex/bin/pex.py [hunk 2]
```diff
@@ -159,7 +159,7 @@ def configure_clp_pex_resolution(parser, builder):
   group.add_option(
       '--cache-dir',
       dest='cache_dir',
-      default=os.path.expanduser('~/.pex/build'),
+      default=os.path.expanduser('{pex_root}/build'),
```
Comment (kwlzn) [inline pex/bin/pex.py]: should this lose the `expanduser` treatment now that we don't specify a string literal with ~ in it?

File: pex/bin/pex.py [hunk 3]
```diff
@@ -266,7 +266,7 @@ def configure_clp_pex_environment(parser):
   group.add_option(
       '--interpreter-cache-dir',
       dest='interpreter_cache_dir',
-      default=os.path.expanduser('~/.pex/interpreters'),
+      default=os.path.expanduser('{pex_root}/interpreters'),
```
Comment (kwlzn) [inline pex/bin/pex.py]: should this also lose the `expanduser` treatment now that we don't specify a string literal with ~ in it?

File: pex/bin/pex.py [hunk 4]
```diff
@@ -338,6 +338,13 @@ def configure_clp():
       help='Turn on logging verbosity, may be specified multiple times.')
 
   parser.add_option(
+    '--pex-root',
```
Comment (kwlzn) [inline pex/bin/pex.py]: nit: this doesn't match the existing indentation style of 4 spaces.

File: pex/bin/pex.py [hunk 5]
```diff
@@ -492,6 +499,10 @@ def build_pex(args, options, resolver_option_builder):
 
 
 def main():
+  def make_relative_to_root(path):
+    """Update options so that defaults are user relative to specified pex_root"""
```
Comment (kwlzn) [inline pex/bin/pex.py]: nit: punctuation

File: pex/bin/pex.py [hunk 6]
```diff
@@ -492,6 +499,10 @@ def build_pex(args, options, resolver_option_builder):
 
 
 def main():
+  def make_relative_to_root(path):
```
Comment (kwlzn) [inline pex/bin/pex.py]: could this get de-nested outside of `main()` to declutter things?

File: pex/variables.py [hunk 7]
```diff
@@ -238,7 +238,7 @@ def PEX_ROOT(self):
     The directory location for PEX to cache any dependencies and code.  PEX must write
     not-zip-safe eggs and all wheels to disk in order to activate them.  Default: ~/.pex
     """
-    return self._get_path('PEX_ROOT', default=None)
+    return self._get_path('PEX_ROOT', default='~/.pex')
```
Comment (kwlzn) [inline pex/variables.py]: should we `os.path.expanduser` this string here?

File: pex/bin/pex.py [hunk 8]
```diff
@@ -503,8 +514,12 @@ def main():
     args, cmdline = args, []
 
   options, reqs = parser.parse_args(args=args)
+  if options.pex_root:
+    ENV.set('PEX_ROOT', options.pex_root)
+  options.cache_dir = make_relative_to_root(options.cache_dir)
+  options.interpreter_cache_dir = make_relative_to_root(options.interpreter_cache_dir)
 
-  with ENV.patch(PEX_VERBOSE=str(options.verbosity)):
+  with ENV.patch(PEX_VERBOSE=str(options.verbosity), PEX_ROOT=options.pex_root):
```
Comment (kwlzn) [inline pex/bin/pex.py]: what happens if `options.pex_root` is `None` here? does that clobber the default `PEX_ROOT`?

File: pex/commands/bdist_pex.py [hunk 9]
```diff
@@ -55,7 +55,7 @@ def run(self):
 
     reqs = [package_dir] + reqs
 
-    with ENV.patch(PEX_VERBOSE=str(options.verbosity)):
+    with ENV.patch(PEX_VERBOSE=str(options.verbosity), PEX_ROOT=options.pex_root):
```
Comment (kwlzn) [inline pex/commands/bdist_pex.py]: what happens if `options.pex_root` is `None` here? does that clobber the default `PEX_ROOT`?

File: pex/bin/pex.py [hunk 10]
```diff
@@ -503,8 +514,12 @@ def main():
     args, cmdline = args, []
 
   options, reqs = parser.parse_args(args=args)
+  if options.pex_root:
```
Comment (kwlzn) [inline pex/bin/pex.py]: should we be doing the inverse here (e.g. set options.pex_root from ENV default) in the case that `options.pex_root` is `None`?

**Comment by kwlzn** [comment]
apologies for the delay in getting to this - have a handful of comments/questions on the PR.

would also like to see some basic test coverage to avoid future regressions.

File: pex/bin/pex.py (hunk 3 above)
Comment (digwanderlust) [inline pex/bin/pex.py]: It is possible for pex root to have ~ in it.

File: pex/bin/pex.py (hunk 2 above)
Comment (digwanderlust) [inline pex/bin/pex.py]: its possible for pex_root to have ~ in it.

**Comment by kwlzn** [comment]
this was abandoned in favor of #206, no?

**Pull request closed without merging**